clients/cli/src/validate.rs

```diff
@@ -0,0 +1,56 @@
+use crate::error::CliResult;
+use crate::utils::{get_account_or_exit, get_config};
+use crate::{err, err_unexpected};
+use lockbook_core::{Error as CoreError, get_path_by_id};
+use lockbook_core::service::integrity_service::{test_repo_integrity, TestRepoError, Warning};
+use uuid::Uuid;
+use lockbook_core::service::{path_service, integrity_service};
+use crate::error::ErrorKind::DocTreatedAsFolder;
+use lockbook_core::model::state::Config;
+
+pub fn validate() -> CliResult<()> {
+    get_account_or_exit();
+
+    let config = get_config();
+
+    match integrity_service::test_repo_integrity(&config) {
+        Ok(warnings) => {
+            // let mut warning_ids = Vec::new();
+            // for w in warnings {
+            //     warning_ids.push(w.uuid);
+            // }
+            // (warning_ids, warnings)
+            err!(DocTreatedAsFolder(String::from("DEBUG")))
+        }
+        Err(err) => match err {
+            TestRepoError::NoRootFolder => err!(NoRootOps("validate")),
+            TestRepoError::DocumentTreatedAsFolder(id) => {
+                err!(DocTreatedAsFolder(get_path_by_id_or_err(&config, id)?))
```

Comment:
There's a few things going on here:

First you do this: `get_path_by_id_or_err(&config, id)?` which either converts or returns early with an error.

Then an error enum is created with that value: `err!(DocTreatedAsFolder(...))`.

Nothing happens with this error though. You can `return` it (not sure if `?` works here you can try that). But what's a more common pattern [for instance in `move`](https://github.com/lockbook/lockbook/blob/master/clients/cli/src/move_file.rs) is to leave this `match` statement as the last line of this `fn`.

Currently your last line is `Ok(())` which is what's causing this value to be ignored.

All you have to do is delete that line, and this code will compile.

Reply:
I actually do get the same compilation error when I remove `Ok(())` -- I'll look into your other suggestions though

Reply:
Hmm, still seeing this last error:

```
error[E0618]: expected function, found enum variant `$crate::error::ErrorKind::$err`
   --> clients/cli/src/error.rs:173:38
    |
45  |             $( $variants $( ( $( $types ),* ) )?, )*
    |                --------- `$crate::error::ErrorKind::$err` defined here
...
173 |         $crate::error::Error::Simple($crate::error::ErrorKind::$err $( ( $( $args ),+ ) )?)
    |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^------------------------
    |                                                                |
    |                                                                call expression requires function
    | 
   ::: clients/cli/src/validate.rs:34:17
    |
34  |                 err!(CycleDetected(get_path_by_id_or_err(&config, id)?))
    |                 -------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: `$crate::error::ErrorKind::$err` is a unit variant, you need to write it without the parenthesis
    |
173 |         $crate::error::Error::Simple($crate::error::ErrorKind::$crate::error::ErrorKind::$err )?)
    |                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```